pkg/api/norman/customization/aks/listers.go

```diff
@@ -326,3 +337,35 @@ func encodeOutput(result interface{}) ([]byte, int, error) {
 
 	return data, http.StatusOK, err
 }
+
+func kubernetesVersionsFromOrchestratorVersionProfiles(profiles ...containerservice.OrchestratorVersionProfile) ([]string, error) {
+	var kubernetesVersions []string
+
+	for _, profile := range profiles {
+		if profile.OrchestratorType == nil || profile.OrchestratorVersion == nil {
+			return kubernetesVersions, fmt.Errorf("unexpected nil orchestrator type or version")
+		}
+
+		if *profile.OrchestratorType == "Kubernetes" {
+			kubernetesVersions = append(kubernetesVersions, *profile.OrchestratorVersion)
+		}
+	}
+
+	return kubernetesVersions, nil
+}
+
+func kubernetesVersionsFromOrchestratorProfiles(profiles ...containerservice.OrchestratorProfile) ([]string, error) {
+	var kubernetesVersions []string
+
+	for _, profile := range profiles {
+		if profile.OrchestratorType == nil || profile.OrchestratorVersion == nil {
```

Comment:
Is the nil check and error return necessary here? Should we just skip over it if this is the case? Could add a debug log if information could be useful.